Add option to require confirmation for external links

johanndev1208 opened the pull request:
## Objectives

- Added a warning banner when users click external links
- Informs users they are leaving our site and data handling is under third-party responsibility
- Implemented to align with GDPR transparency and information duties
  (Art. 5(1)(a), Art. 12–14, Recital 39)

## Visual Changes


https://github.com/user-attachments/assets/44d106d4-2706-4cf0-915e-271b22048b63


<img width="1182" height="501" alt="Screenshot 2025-09-25 at 20 38 55" src="https://github.com/user-attachments/assets/53b1d62e-7cdc-40ff-b70a-7881243ae166" />

johanndev1208 committed:
Require confirmation when clicking external links

While there is no explicit provision in the GDPR that requires this in
exactly this way, for the demokratie.today version a data protection
specialist (lawyer) was consulted, and they recommended including this
warning based on the German legal interpretation.

javierm commented:
Hi, @itw-smart :smile:.

Understood, no problem! :+1:

Just to clarify, my intention wasn't doing an additional legal review; very sorry it sounded like that! :pray: The thing is that we (maintainers) are expected` to know the reason why any part of the application is done a certain way. So, in that sense, your comment totally makes a difference :smile:. If people ask us about this feature, we can now say something like "some countries make a legal interpretation of the GDPR that requires this feature".

javierm committed:
Move footer CSS to its own stylesheets

Like we usually do, the stylesheets are named after the components.

javierm committed:
Add an icon to external links in the footer

For now it's only in the footer because it's the only place where we use
`rel=external`. We might add it to external links in user-generated
content in the future.

taitus committed:
Remove redundant url option in notifications spec

Capybara defaults to url: true when the expected path is a full URL.
